nemo/collections/nlp/models/machine_translation/mt_enc_dec_bottleneck_model.py

```diff
@@ -49,51 +62,38 @@ class MTBottleneckModel(MTEncDecModel):
     def __init__(self, cfg: MTBottleneckModelConfig, trainer: Trainer = None):
         super().__init__(cfg=cfg, trainer=trainer)
 
-        recon_per_token: bool = True
-
-        self.model_type: str = cfg.get("model_type", "seq2seq-br")
+        self.model_type: str = cfg.get("model_type", "recon_only")
         self.min_logv: float = cfg.get("min_logv", -6)
-        self.ortho_loss_coef: float = cfg.get("ortho_loss_coef", 0.0)
-        self.att_bridge_size: int = cfg.get("att_bridge_size", 512)
-        self.att_bridge_k: int = cfg.get("att_bridge_k", 16)
-        self.att_bridge_inner_size: int = cfg.get("att_bridge_inner_size", 1024)
+        self.latent_size: int = cfg.get("latent_size", -1)
         self.non_recon_warmup_batches: int = cfg.get("non_recon_warmup_batches", 200000)
         self.recon_per_token: bool = cfg.get("recon_per_token", True)
 
-        # TODO: add support in label smoothing for per-sample reconstruction loss
+        # latent_size -1 will take value of encoder.hidden_size
+        if self.latent_size < 0:
+            self.latent_size = self.encoder.hidden_size
+
+        # remove me
+        self.att_bridge_k: int = cfg.get("att_bridge_k", 16)
+        self.att_bridge_inner_size: int = cfg.get("att_bridge_inner_size", 1024)
+
+        # TODO: add label smoothing support for per-sample reconstruction loss
         if not self.recon_per_token:
             loss_fn = NLLLoss(ignore_index=self.decoder_tokenizer.pad_id, reduction='none',)
             self.loss_fn = self.eval_loss_fn = loss_fn
 
-        if self.model_type not in ["seq2seq", "seq2seq-br", "seq2seq-mim", "seq2seq-vae"]:
+        if self.model_type not in ["recon_only", "mim", "vae"]:
             raise ValueError("Unknown model_type = {model_type}".format(model_type=self.model_type,))
```

Comment:
Just a nitpick, but you could use an f-string here:
`f"Unknown model type: {self.model_type}"`

Reply:
Done

Reply:
Done.